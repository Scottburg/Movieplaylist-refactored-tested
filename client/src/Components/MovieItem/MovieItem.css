.playlistButton {
  margin:  10px 0 20px 0;
  width: 200px;
}

.singleMovie {
  max-width: 300px;
  margin-right: 10px;
  margin-left: 10px;
  margin-bottom: 40px;
}

.poster-img {
  width: 200px;
  height: 300px;
  border-radius: 5px;
  box-shadow: 2px 2px 2px 2px rgba(0, 0, 0, 0.2);
}
