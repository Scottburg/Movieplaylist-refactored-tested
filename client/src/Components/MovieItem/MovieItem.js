import React, { useState } from 'react';
import ListOfSongs from '../../Containers/SongList/SongList';
import Button from 'react-bootstrap/Button';
import noPosterImg from '../../images/no-poster-available.png';
import './MovieItem.css';

function MovieItem({ title, posterPath, date }) {
  const [showPlaylist, setShowPlaylist] = useState(false);
  const showingButtonText = showPlaylist ? 'Hide Playlist' : 'Show Playlist';

  const openPlaylist = () => {
    return setShowPlaylist(!showPlaylist);
  };

  return (
    <div className="singleMovie">
      <div>
        <span>
          <h5 data-testid="title">{title}</h5>
        </span>
        <img className="poster-img"
          style={{ width: '200px', height: '300px' }}
          alt="movie post"
          src={
            posterPath
              ? `http://image.tmdb.org/t/p/w200${posterPath}`
              : noPosterImg
          }
        />
        <Button className="playlistButton"
          variant="outline-primary"
          onClick={openPlaylist}
        >
          {showingButtonText}
        </Button>
      </div>
      {showPlaylist && <ListOfSongs title={title} />}
    </div>
  );
}

export default MovieItem;
