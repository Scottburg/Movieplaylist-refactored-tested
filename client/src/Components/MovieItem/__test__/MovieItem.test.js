import React from 'react';
import MovieItem from '../MovieItem';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

// import react-testing methods
import { render, cleanup } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom/extend-expect';

// mocking Redux Provider and store
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import allReducers from '../../../Reducers';
const initialState = {
  authentication: {
    spotifyToken: 'DDD',
    spotifyId: 'DDD',
  },
  movieList: false,
};
const store = createStore(allReducers, initialState);

afterEach(cleanup);

it('should take a snapshot', () => {
  const wrapper = (
    <Provider store={store}>
      <MovieItem></MovieItem>
    </Provider>
  );
  const { asFragment } = render(wrapper);

  expect(asFragment(wrapper)).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MovieItem />, div);
});

it('should be defined', () => {
  expect(MovieItem).toBeDefined();
});

it('has a title', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <MovieItem />
    </Provider>
  );
  expect(getByTestId('title').textContent).toBeInTheDocument;
});

it('changes text when button clicked', () => {
  const wrapper = (
    <Provider store={store}>
      <MovieItem></MovieItem>
    </Provider>
  );
  const { getByText } = render(wrapper);
  const button = getByText(/show playlist/i);
  expect(button).toHaveTextContent(/show playlist/i);

  button.click();
  expect(button).toHaveTextContent(/hide playlist/i);
});
