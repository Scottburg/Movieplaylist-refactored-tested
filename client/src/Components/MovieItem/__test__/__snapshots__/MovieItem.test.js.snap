// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="singleMovie"
  >
    <div>
      <span>
        <h5
          data-testid="title"
        />
      </span>
      <img
        alt="movie post"
        class="poster-img"
        src="no-poster-available.png"
        style="width: 200px; height: 300px;"
      />
      <button
        class="playlistButton btn btn-outline-primary"
        type="button"
      >
        Show Playlist
      </button>
    </div>
  </div>
</DocumentFragment>
`;
